src/constraints/constraint_connection_flow_gas_capacity.jl

```diff
@@ -0,0 +1,85 @@
+#############################################################################
+# Copyright (C) 2017 - 2018  Spine Project
+#
+# This file is part of SpineOpt.
+#
+# SpineOpt is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# SpineOpt is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+"""
+    add_constraint_connection_flow_gas_capacity!(m::Model)
+
+This constraint is needed to force uni-directional flow over gas connections.
+"""
+function add_constraint_connection_flow_gas_capacity!(m::Model)
+    @fetch connection_flow,binary_gas_connection_flow = m.ext[:variables]
+    m.ext[:constraints][:connection_flow_gas_capacity] = Dict(
+    (connection=conn, node1=n_from, node2=n_to, stochastic_scenario=s,t=t) => @constraint(
+        m,
+        (sum(
+        connection_flow[conn, n_from, d, s, t]*duration(t)
+        for (conn,n_from,d,s,t) in connection_flow_indices(m;connection=conn, node=n_from, stochastic_scenario=s, t=t_in_t(m;t_long=t),direction=direction(:from_node))
+        )
+        +
+        sum(
+        connection_flow[conn, n_to, d, s, t]*duration(t)
+        for (conn,n_to,d,s,t) in connection_flow_indices(m;connection=conn, node=n_to, stochastic_scenario=s, t=t_in_t(m;t_long=t),direction=direction(:to_node))
+        )) /2
+        <=
+        + bigM(model=m.ext[:instance])
+        *
+        sum(
+        binary_gas_connection_flow[conn, n_to, d, s, t]*duration(t)
+        for (conn,n_to,d,s,t) in connection_flow_indices(m;connection=conn, node=n_to, stochastic_scenario=s, t=t_in_t(m;t_long=t),direction=direction(:to_node))
+        )
+        ) for (conn,n_from,n_to,s,t) in constraint_connection_flow_gas_capacity_indices(m)
+        )
+end

```

Comment:
There's some funny indentations here, and missing spaces after commas. If you find the time it would be good to fix it, but also ok if not.